### DISCUSSION TITLE
Not clear how to interop with Java collections and other objects

### DISCUSSION THREAD

**Discussion by chrisseaton** [oracle/truffleruby#1552] [General] [answered]
@eropple we'll open this issue to discuss what you're trying to do with interop.

I'll give you some background. We're probably not aiming short term to prioritise matching JRuby's interop. We're building a single interop mechanism that works with any language, not just Java. That means it doesn't pretend that foreign objects as Ruby objects in exactly the same way, as we want it to be a bit more uniform and explicit.

Could you please re-state here what you want to do and we'll talk you through the several options there are. We're really interested in your use-case because we're to some extent not sure what's best ourselves yet and we're hoping that actual use-cases might make the best option more clear.

I'm also generally interested in your application which you say is running well on TruffleRuby. Is that publicly accessible code we could run ourselves?

**Comment by eregon** [comment, marked as answer]
This is documented in https://github.com/oracle/truffleruby/blob/master/doc/user/jruby-migration.md#using-java-from-ruby.
